### ISSUE TITLE
[Feature request] Create command to use version for one command

### ISSUE THREAD

**Issue by deiga** [asdf-vm/asdf#613]
Would it be possible to get a command like `asdf use <plugin> <version> <command>`?

It would be great to run one-off commands against a specific version

**Comment by vic** [comment]
You can `env ASDF_FOO_VERSION=1.0 foo` and the `foo` shim will determine the version from the environment variable.

**Comment by deiga** [comment]
Yeah, that's true. For my case it's also good enough. But I wonder if this addition wouldn't still be useful. Other version management tools have similar functionality as well, so I would guess that it could be a sought after feature

**Comment by Stratus3D** [comment]
I think `asdf shell <plugin> <version>` should be able to do what you need.

Something like:

```
$(asdf shell ruby 2.6.3; ruby --version)
```

If your goal is a convenience command, a simple shell function should get you what you need:

```
asdf-command() {
  plugin=$1
  version=$2
  command=$3

  $(asdf shell $plugin $version; $command)
}
```

I don't think this is something we should add to asdf.

**Comment by Stratus3D** [comment]
Closing since the original feature request is not going to be implemented.

**Comment by blowfishpro** [comment]
I got this script working.  There are probably some simplifications that could be made and edge cases I didn't handle but it's doing what I want for now:
```bash
#!/usr/bin/env bash -e

asdf_args=()

for var in "${@}"
do
  if [[ "${var}" == '--' ]]
  then
    shift
    break
  else
    asdf_args+=("${var}")
    shift
  fi
done

if [[ "${#}" -eq 0 ]] || [[ "${#asdf_args[@]}" -eq 0 ]]
then
  echo "usage: ${0##*/} [asdf_args...] -- [command...]" >&2
  exit 1
fi

. /usr/local/opt/asdf/asdf.sh
asdf shell "${asdf_args[@]}"
exec "${@}"
```

It is sort of nice when these things are built into the version manager, it seems to be a pretty polarizing topic though as to whether they should be included or not.
